public/js/redux/chartsRedux.js

```diff
@@ -2,23 +2,47 @@ import { State, Actions, Effect } from 'jumpstate';
 import api from 'services/Api';
 import chartList from 'utils/chartList';
 import { createYTotalsList, createPartialsList, formattedSeries, compareDates, fillMissingDates } from 'helpers/chartsHelpers';
+import { reEstablishSession } from 'panda-session';
 import moment from 'moment';
 
 /* ------------- State Management ------------- */
 
+const updateAttemptActions = (chartData, chart, startDate, endDate) => {
+    Actions.toggleIsUpdatingCharts(false);
+    Actions[`update${chart}`]({ chartData, startDate, endDate });
+};
+
+const responseFailActions = (chart, error) => {
+    Actions.toggleIsUpdatingCharts(false);
+    Actions[`get${chart}Failed`](error);
+};
+
 Effect('filterDesk', (filterObj) => {
     Actions.updateFilter(filterObj);
     Actions.toggleIsUpdatingCharts(true);
     const { startDate, endDate, desk, productionOffice } = filterObj;
     chartList.map(chart => {
         api[`get${chart}`](startDate, endDate, desk, productionOffice)
-            .then(chartData => {
-                Actions.toggleIsUpdatingCharts(false);
-                Actions[`update${chart}`]({ chartData, startDate, endDate });
-            })
+            .then(chartData => updateAttemptActions(chartData, chart, startDate, endDate))
             .catch(error => {
-                Actions.toggleIsUpdatingCharts(false);
-                Actions[`get${chart}Failed`](error);
+                const status = error.response.status;
+                if (status === 419) {
```

Comment:
I would argue that this if statement lives in a separate function as the same logic will be required for every call to an auth-ed endpoint

Reply:
Good point!

Reply:
Good point!